Makefile

```diff
@@ -589,6 +603,30 @@ $(DMLC_CORE)/libdmlc.a: DMLCCORE
 DMLCCORE:
 	+ cd $(DMLC_CORE); $(MAKE) libdmlc.a USE_SSE=$(USE_SSE) config=$(ROOTDIR)/$(config); cd $(ROOTDIR)
 
+TVM_USE_CUDA := OFF
```

Comment:
Is it really necessary to introduce another variable? Couldn't we just keep USE_CUDA and tie onto that instead?

Reply:
well, what I am trying to do is to attach `-DUSE_CUDA=$USE_CUDA_PATH` to TVM if `USE_CUDA_PATH` is set, otherwise attach `-DUSE_CUDA=$USE_CUDA`. I think it is fine as it is not exposed to users.

Reply:
Could we have that logic by checking use_cuda then instead? I see your point and it's certainly simpler, but we have a lot of diverging logic branches around the use_cuda flag already, so I'd like to avoid further branching by introducing additional variables that branch off of that.

Reply:
you mean ignore `USE_CUDA_PATH` ?

Reply:
No, what I mean is that the TVM related logic should be in a single place which infers from our base variables USE_TVM, USE_CUDA and USE_CUDA_PATH. Any variables that are now TVM specific should be constructed in a single block that has "if USE_TVM" as condition. These variables don't leave the scope of that block.

Right now, this introduces more and more TVM specific variables on a global level and scattered across the makefile. Instead it would be good if we have a single block to invoke TVM - this block will take care of converting all existing variables to their TVM counterpart. This improves the modularity and acts as a clear separation between our build system and TVM. It acts like a "translator". The issue to me here is that we are mirroring existing variables we already have at the global scope. Combining this fact with conditional operations on top of these mirrored variables, makes it harder to understand the process flow.

I hope that this makes it clearer

Reply:
Just look how you did it in cmake. It's a single and self enclosed block. I'd like to see the same here

Reply:
addressed, does it look better now?

Reply:
Looks great, thank you!

Optional: Indentation (shift them one tab to the right because of the if-block) - but I'll leave it up to you :)

Reply:
opinion of aesthetics  :)